### PR TITLE
WIP: Add metrics to engine

### PR DESCRIPTION
**Pull request by PradyumnaKrishna**
**Description**
Adds metrics to thanos promql engine.

**Implemented metrics**
* `currentQueries`: View no of queries executing through thanos promql engine.

### CODE REVIEW

File: engine/engine.go [hunk 1]
```diff
@@ -129,6 +133,19 @@ func New(opts Opts) *compatibilityEngine {
...
+		}),
+	}
+
+	if opts.EngineOpts.Reg != nil {
+		opts.EngineOpts.Reg.MustRegister(metrics.currentQueries)
+	}
+
 	return &compatibilityEngine{
 		prom: promql.NewEngine(opts.EngineOpts),
 		queries: promauto.With(opts.Reg).NewCounterVec(
```
Comment (GiedriusS) [inline engine/engine.go:151]: Maybe this can be moved into the new `engineMetrics`?

File: engine/engine.go [hunk 2]
```diff
@@ -129,6 +133,19 @@ func New(opts Opts) *compatibilityEngine {
...
+		currentQueries: prometheus.NewGauge(prometheus.GaugeOpts{
+			Namespace: "thanos",
+			Subsystem: "engine",
+			Name:      "queries",
+			Help:      "The current number of queries being executed or waiting.",
+		}),
+	}
+
+	if opts.EngineOpts.Reg != nil {
+		opts.EngineOpts.Reg.MustRegister(metrics.currentQueries)
```
Comment (GiedriusS) [inline engine/engine.go:146]: Let's use `promauto.With(opts.Reg)`? It seems like `opts.Reg` is a mandatory parameter.

File: engine/engine.go [hunk 3]
```diff
@@ -129,26 +136,38 @@ func New(opts Opts) *compatibilityEngine {
...
+			prometheus.GaugeOpts{
+				Namespace: namespace,
+				Subsystem: subsystem,
+				Name:      "queries",
+				Help:      "The current number of queries being executed or waiting.",
+			},
+		),
 		queries: promauto.With(opts.Reg).NewCounterVec(
 			prometheus.CounterOpts{
 				Name: "promql_engine_queries_total",
```
Comment (saswatamcode) [inline engine/engine.go:150]: I wonder if the name should be changed to `thanos_engine_queries_total` as well? Wdyt @GiedriusS?

Comment (GiedriusS) [reply engine/engine.go:150]: Makes sense, we can re-use the same pattern as in `currentQueries` :+1:

**Review by GiedriusS — APPROVED** [review]

**Review by saswatamcode — APPROVED** [review]

**Pull request merged**